pkg/sync/mutex_dev.go

```diff
@@ -86,7 +86,7 @@ func (m *Mutex) TryLock() bool {
 // Unlock releases an acquired lock on the mutex.
 func (m *Mutex) Unlock() {
 	panicIfTooMuchTimeElapsed("Mutex.Unlock", m.acquireTime, lockTimeout, 1)
-	m.Mutex.Unlock()
+	defer m.Mutex.Unlock() // suppress the error for calling Unlock()
```

Comment:
How about moving it up and removing comment

Reply:
This changes the logic, right? I'm not sure the intent here is to call `Unlock()` in case the previous function panic'd.